client/index.js

```diff
@@ -8,17 +8,35 @@ import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
 // check for existing cards
 
 function App() {
+  const movieGridRef = createRef();
+  const [searchQuery, setSearchQuery] = useState('');
+
+  useEffect(() => {
+  }, [movieGridRef]);
+
+
+  function searchTopMovies() {
+    movieGridRef.current.searchTopMovies();
```

Comment:
ah IC, calls into the movieGridRef to access its searchTopMovies. 🤔

Reply:
So I see what's going on here, we don't have a global store like Redux (or even context) to store this data, so we're letting the MovieGrid own it, and then we have to do some gymnastics to make the MovieGrid be clickable

I agree, having the Button be in the MovieGrid would definitely be easier (and certainly appropriate for the purposes of this exercise) - 

I might also suggest moving the MovieData ownership up to the top-level component (`App`) and passing the data down to the MovieGrid component. That way the App can know "everything" and act as our global Store, and we keep our uni-directional data flow. Thoughts? Give it a try, I think you might be pleased with what comes out

(The principle behind this: https://en.wikipedia.org/wiki/Inversion_of_control - when in doubt, flip the caller and the object, ha) 🙌